packages/sdk/src/sdk/api/emails/EmailsApi.ts

```diff
@@ -0,0 +1,188 @@
+          granteeSharedSecret,
+          symmetricKey
+        )
+        return {
+          granteeId,
+          encryptedKey: base64.encode(encryptedKeyBytes)
+        }
+      })
+    )
+
+    return {
+      ownerEncryptedKey,
+      granteeEncryptedKeys,
+      symmetricKey
+    }
+  }
+
+  /**
+   * Decrypts the symmetric key for either an owner or grantee
+   * @param encryptedKey The encrypted symmetric key as a base64 string
+   * @param userId The ID of the user who encrypted the key
+   * @returns The decrypted symmetric key
+   */
+  async decryptSymmetricKey(
+    encryptedKey: string,
+    userId: string
+  ): Promise<Uint8Array> {
+    const userPublicKey = await this.getPublicKey(userId)
+    const sharedSecret = await this.auth.getSharedSecret(userPublicKey)
+    return await CryptoUtils.decrypt(sharedSecret, base64.decode(encryptedKey))
+  }
+
+  /**
+   * Encrypts an email using a symmetric key
+   * @param email The email to encrypt
+   * @param symmetricKey The symmetric key to use for encryption
+   * @returns The encrypted email as a base64 string
+   */
+  async encryptEmail(email: string, symmetricKey: Uint8Array): Promise<string> {
+    const encryptedBytes = await CryptoUtils.encryptString(symmetricKey, email)
+    return base64.encode(encryptedBytes)
+  }
+
+  /**
+   * Decrypts an email using a symmetric key
+   * @param encryptedEmail The encrypted email as a base64 string
+   * @param symmetricKey The symmetric key to use for decryption
+   * @returns The decrypted email as a string
+   */
+  async decryptEmail(
+    encryptedEmail: string,
+    symmetricKey: Uint8Array
+  ): Promise<string> {
+    return await CryptoUtils.decryptString(
+      symmetricKey,
+      base64.decode(encryptedEmail)
+    )
+  }
+
+  /**
+   * Encrypts emails for a seller and their grantees
+   * @param sellerId The ID of the seller
+   * @param granteeIds List of grantee user IDs
+   * @param emails List of emails to encrypt
+   * @returns Object containing encrypted emails and encrypted symmetric keys for both seller and grantees
+   */
+  async encryptEmails(
+    sellerId: string,
+    granteeIds: string[],
+    emails: string[]
+  ): Promise<EncryptedEmailsResult> {
+    // Create symmetric key for seller and grantees
+    const { symmetricKey, ownerEncryptedKey, granteeEncryptedKeys } =
+      await this.createSharedKey(sellerId, granteeIds)
+
+    // Encrypt emails with symmetric key
+    const encryptedEmails = await Promise.all(
+      emails.map((email) => this.encryptEmail(email, symmetricKey))
+    )
+
+    return {
+      encryptedEmails,
+      ownerKey: ownerEncryptedKey,
+      granteeKeys: granteeEncryptedKeys
+    }
+  }
+
+  /**
+   * Batch encrypts emails for multiple sellers and their grantees
+   * @param data Array of objects containing seller IDs, grantee IDs, and emails to encrypt
+   * @returns Array of encryption results for each seller
+   */
+  async batchEncryptEmails(
```

Comment:
only gonna be used for backfilling, not sure if we should add here or not

Reply:
I think it's fine to have here, we can rip it out later if needed. Why the mixed casings though? We should stick to camelCase if we can?

Reply:
just what we get from running the sql query from the dbs, we'll rip this out later as you said, just keeping here for future backfill usage